homeassistant/components/sonos/media_player.py

```diff
@@ -1073,22 +1095,21 @@ def restore(entities, with_group):
         if entity.state == STATE_PLAYING:
             entity.media_pause()
 
-    # Bring back the original group topology and clear pysonos cache
+    # Bring back the original group topology
     if with_group:
         for entity in (e for e in entities if e._snapshot_group):
             if entity._snapshot_group[0] == entity:
                 entity.join(entity._snapshot_group)
-                entity.soco._zgs_cache.clear()
 
-    # Restore slaves, then coordinators
-    slaves = [e for e in entities if not e.is_coordinator]
-    coordinators = [e for e in entities if e.is_coordinator]
-    for entity in slaves + coordinators:
-        try:
-            entity._soco_snapshot.restore()
-        except (TypeError, AttributeError, SoCoException) as ex:
-            # Can happen if restoring a coordinator onto a current slave
-            _LOGGER.warning("Error on restore %s: %s", entity.entity_id, ex)
+    # Restore slaves
+    for entity in (e for e in entities if not e.is_coordinator):
+        entity.restore()
 
-        entity._soco_snapshot = None
-        entity._snapshot_group = None
+    # Restore coordinators (or delay if moving from slave)
+    for entity in (e for e in entities if e.is_coordinator):
+        if entity._sonos_group[0] == entity:
+            # Was already coordinator
+            entity.restore()
+        else:
+            # Await coordinator role
+            entity._restore_pending = True
```

Comment:
Since it's used outside of the class, it should not be prefixed with an `_` ?

Reply:
It's more that I think static methods are ugly in Python ... but, I moved them inside the class now 😃

Reply:
That solve the problem not that we need disable protected access?

Reply:
Unfortunately not, pylint still flagged it when accessing with `entity` (i.e. not `self`).

I am actually unsure if that is a shortcoming of pylint or maybe the Python convention is not like C++ ? (where static methods do have access to private members)